src/compiler/checker.ts

```diff
@@ -4022,6 +4029,7 @@ namespace ts {
                     type.typeParameters = concatenate(outerTypeParameters, localTypeParameters);
                     type.outerTypeParameters = outerTypeParameters;
                     type.localTypeParameters = localTypeParameters;
+                    type.minTypeArgumentCount = getMinTypeArgumentCount(localTypeParameters);

```

Comment:
Considering the number of times we need to check the arity of type arguments, it seemed worth the effort to pre-calculate this value just as we do `minArgumentCount` on signatures. Even if we compute on demand, I would likely want to cache the result to avoid repeatedly computing this value.